packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -588,6 +590,20 @@ class SliderThemeData with Diagnosticable {
   /// If specified, overrides the default value of [Slider.allowedInteraction].
   final SliderInteraction? allowedInteraction;
 
+  /// The size of the [BarSliderThumbShape] thumb.
+  ///
+  /// If [SliderThemeData.thumbShape] is [BarSliderThumbShape], this property is used to
+  /// set the size of the thumb. Otherwise, the default thumb size is 4 pixels for the
+  /// width and 44 pixels for the height.
+  final MaterialStateProperty<Size?>? barThumbSize;
```

Comment:
The reason i made it `barThumbSize` because it's a `Size` type and this is not compatible with existing Thumb as it is just a circle and it would be `double` aka radius value. If the user expect `thumbSize` to update existing thumb it might create confusion. 

However, you rasied a good point on consistency. I will take a look if we `Size` `thumbSize` can be used for existing thumb radius by rounding to square value.

Reply:
I agree. I did not realize this theme data class was so large. There are a lot of parameters here, so having a clear name would be ideal.

Reply:
Maybe (in a separate change) we can look at this, and maybe group the parameters together based on the type of slider they apply to.